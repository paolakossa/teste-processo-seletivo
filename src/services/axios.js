import axios from 'axios';
import { store } from '../store/store';

axios.interceptors.request.use(
  (config) => {
    const { token } = store.getState().userData;
    config.baseURL = 'https://reqres.in/api';

    if (token && config.headers) {
      config.headers.Authorization = `Bearer ${token}`;
    }
    return config;
  },
  () => {
    return Promise.reject('Ocorreu um erro !');
  },
);

export default axios;
