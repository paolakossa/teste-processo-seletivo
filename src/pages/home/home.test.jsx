import React from 'react';
import { render, screen } from '@testing-library/react';
import { describe, it, vi, beforeEach, expect } from 'vitest';
import axios from 'axios';
import { Home } from './index';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { configureStore } from '@reduxjs/toolkit';
import userDataReducer from '../../store/auth/authSlice';

vi.mock('axios');

const createTestStore = (preloadedState) =>
  configureStore({
    reducer: {
      userData: userDataReducer,
    },
    preloadedState,
  });

const renderHome = (state = { userData: { token: 'fake-token', email: 'teste@email.com' } }) => {
  const store = createTestStore(state);
  const queryClient = new QueryClient();

  return render(
    <Provider store={store}>
      <MemoryRouter>
        <QueryClientProvider client={queryClient}>
          <Home />
        </QueryClientProvider>
      </MemoryRouter>
    </Provider>,
  );
};

describe('Home Page', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('deve mostrar o spinner de carregamento', async () => {
    axios.get.mockReturnValue(new Promise(() => {}));
    renderHome();

    expect(document.querySelector('.ant-spin')).toBeInTheDocument();
  });

  it('deve exibir usuários após carregamento', async () => {
    axios.get.mockResolvedValueOnce({
      data: {
        data: [
          {
            id: 1,
            first_name: 'Maria',
            last_name: 'Silva',
            email: 'maria@email.com',
            avatar: 'https://via.placeholder.com/40',
          },
        ],
      },
    });

    renderHome();

    const userName = await screen.findByText(/Maria Silva/i);
    expect(userName).toBeInTheDocument();
    expect(screen.getByText(/maria@email.com/i)).toBeInTheDocument();
  });

  it('deve bater com snapshot', async () => {
    axios.get.mockResolvedValueOnce({
      data: {
        data: [
          {
            id: 1,
            first_name: 'Maria',
            last_name: 'Silva',
            email: 'maria@email.com',
            avatar: 'https://via.placeholder.com/40',
          },
        ],
      },
    });

    const { container } = renderHome();
    await screen.findByText(/Maria Silva/i);
    expect(container).toMatchSnapshot();
  });
});
