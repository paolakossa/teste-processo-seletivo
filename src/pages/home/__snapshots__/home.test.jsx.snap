// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Page > deve bater com snapshot 1`] = `
.c1 {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: #fff;
  padding: 0 1rem;
  border-bottom: 1px solid #ddd;
  height: 64px;
}

.c2 {
  font-weight: bold;
  font-size: 1.2rem;
}

.c3 {
  display: flex;
  align-items: center;
  gap: 8px;
  cursor: pointer;
}

.c4 {
  transition: transform 0.3s ease-in-out;
}

.c7 {
  margin: 1rem 0;
  width: 328px;
}

.c8 .ant-input-search-button {
  border-start-end-radius: 8px;
  border-end-end-radius: 8px;
}

.c9 {
  margin-top: 1rem;
  overflow-x: auto;
}

.c10 {
  width: 100%;
}

.c10 .ant-table {
  width: 100%;
  min-width: 600px;
}

.c10 .ant-table-thead>tr>th {
  border: none!important;
  border-bottom: 1px solid #d9d9d9!important;
}

.c10 .ant-table-thead>tr>th::before,
.c10 .ant-table-thead>tr>th::after {
  display: none!important;
  content: none!important;
}

.c10 .ant-table-column-sorters {
  border: none!important;
  background: transparent!important;
  box-shadow: none!important;
}

.c10 .ant-table-thead>tr>th.gray-bg {
  background-color: #f0f0f0!important;
}

.c10 .ant-table-thead>tr>th.gray-bg:hover,
.c10 .ant-table-thead>tr>th.gray-bg.ant-table-column-sort {
  background-color: #f0f0f0!important;
}

.c11 {
  display: flex;
  justify-content: center;
  margin-top: 1rem;
}

.c0 {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  background-color: #f7f9fa;
}

.c5 {
  flex: 1;
  padding: 2.5rem 2rem;
  display: flex;
  flex-direction: column;
  gap: 1rem;
}

.c6 {
  font-weight: 600;
  font-size: 1.5rem;
  font-family: 'Sora',sans-serif;
}

@media (max-width: 768px) {
  .c10 .ant-table-thead>tr>th {
    font-size: 14px;
    padding: 8px;
  }
}

<div>
  <div
    class="ant-layout c0 css-dev-only-do-not-override-tjsggz"
  >
    <header
      class="c1"
    >
      <div
        class="c2"
      >
        ITA Frotas
      </div>
      <div
        class="c3 ant-dropdown-trigger"
      >
        <span>
          teste@email.com
        </span>
        <span
          aria-label="down"
          class="anticon anticon-down c4"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </span>
      </div>
    </header>
    <main
      class="ant-layout-content c5 css-dev-only-do-not-override-tjsggz"
    >
      <h2
        class="c6"
      >
        Usu√°rios
      </h2>
      <div
        class="c7"
      >
        <span
          class="ant-input-group-wrapper ant-input-group-wrapper-outlined css-dev-only-do-not-override-tjsggz ant-input-search ant-input-search-with-button c8"
        >
          <span
            class="ant-input-wrapper ant-input-group css-dev-only-do-not-override-tjsggz"
          >
            <input
              class="ant-input css-dev-only-do-not-override-tjsggz ant-input-outlined"
              placeholder="Pesquise nome ou documento"
              type="search"
              value=""
            />
            <span
              class="ant-input-group-addon"
            >
              <button
                class="ant-btn css-dev-only-do-not-override-tjsggz ant-btn-primary ant-btn-color-primary ant-btn-variant-solid ant-input-search-button"
                type="button"
              >
                <span
                  class="ant-btn-icon"
                >
                  <span
                    aria-label="search"
                    class="anticon anticon-search"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="search"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
            </span>
          </span>
        </span>
      </div>
      <div
        class="c9"
      >
        <div
          class="ant-table-wrapper c10 css-dev-only-do-not-override-tjsggz"
        >
          <div
            class="ant-spin-nested-loading css-dev-only-do-not-override-tjsggz"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table css-dev-only-do-not-override-tjsggz"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                  >
                    <table
                      style="table-layout: auto;"
                    >
                      <colgroup>
                        <col
                          style="width: 5%;"
                        />
                        <col
                          style="width: 30%;"
                        />
                        <col
                          style="width: 25%;"
                        />
                        <col
                          style="width: 20%;"
                        />
                      </colgroup>
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            ID
                          </th>
                          <th
                            aria-label="Nome completo"
                            class="ant-table-cell gray-bg ant-table-column-has-sorters"
                            scope="col"
                            tabindex="0"
                          >
                            <div
                              aria-describedby="test-id"
                              class="ant-table-column-sorters"
                            >
                              <span
                                class="ant-table-column-title"
                              >
                                Nome completo
                              </span>
                              <span
                                class="ant-table-column-sorter ant-table-column-sorter-full"
                              >
                                <span
                                  aria-hidden="true"
                                  class="ant-table-column-sorter-inner"
                                >
                                  <span
                                    aria-label="caret-up"
                                    class="anticon anticon-caret-up ant-table-column-sorter-up"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="caret-up"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="0 0 1024 1024"
                                      width="1em"
                                    >
                                      <path
                                        d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                      />
                                    </svg>
                                  </span>
                                  <span
                                    aria-label="caret-down"
                                    class="anticon anticon-caret-down ant-table-column-sorter-down"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="caret-down"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="0 0 1024 1024"
                                      width="1em"
                                    >
                                      <path
                                        d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </span>
                            </div>
                          </th>
                          <th
                            aria-label="E-mail"
                            class="ant-table-cell gray-bg ant-table-column-has-sorters"
                            scope="col"
                            tabindex="0"
                          >
                            <div
                              aria-describedby="test-id"
                              class="ant-table-column-sorters"
                            >
                              <span
                                class="ant-table-column-title"
                              >
                                E-mail
                              </span>
                              <span
                                class="ant-table-column-sorter ant-table-column-sorter-full"
                              >
                                <span
                                  aria-hidden="true"
                                  class="ant-table-column-sorter-inner"
                                >
                                  <span
                                    aria-label="caret-up"
                                    class="anticon anticon-caret-up ant-table-column-sorter-up"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="caret-up"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="0 0 1024 1024"
                                      width="1em"
                                    >
                                      <path
                                        d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                      />
                                    </svg>
                                  </span>
                                  <span
                                    aria-label="caret-down"
                                    class="anticon anticon-caret-down ant-table-column-sorter-down"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="caret-down"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="0 0 1024 1024"
                                      width="1em"
                                    >
                                      <path
                                        d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </span>
                            </div>
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            A√ß√µes
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="1"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            1
                          </td>
                          <td
                            class="ant-table-cell gray-bg"
                          >
                            <div
                              class="ant-space css-dev-only-do-not-override-tjsggz ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
                            >
                              <div
                                class="ant-space-item"
                              >
                                <span
                                  class="ant-avatar ant-avatar-circle ant-avatar-image css-dev-only-do-not-override-tjsggz"
                                >
                                  <img
                                    src="https://via.placeholder.com/40"
                                  />
                                </span>
                              </div>
                              <div
                                class="ant-space-item"
                              >
                                <span>
                                  Maria
                                   
                                  Silva
                                </span>
                              </div>
                            </div>
                          </td>
                          <td
                            class="ant-table-cell gray-bg"
                          >
                            maria@email.com
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <a
                              href="#"
                            >
                              Mais detalhes
                            </a>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c11"
      >
        <ul
          class="ant-pagination css-dev-only-do-not-override-tjsggz"
        >
          <li
            class="ant-pagination-total-text"
          >
            Total 1 usu√°rios
          </li>
          <li
            aria-disabled="true"
            class="ant-pagination-prev ant-pagination-disabled"
            title="Previous Page"
          >
            <button
              class="ant-pagination-item-link"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                aria-label="left"
                class="anticon anticon-left"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="left"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                  />
                </svg>
              </span>
            </button>
          </li>
          <li
            class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
            tabindex="0"
            title="1"
          >
            <a
              rel="nofollow"
            >
              1
            </a>
          </li>
          <li
            aria-disabled="true"
            class="ant-pagination-next ant-pagination-disabled"
            title="Next Page"
          >
            <button
              class="ant-pagination-item-link"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                aria-label="right"
                class="anticon anticon-right"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </button>
          </li>
          <li
            class="ant-pagination-options"
          >
            <div
              aria-label="Page Size"
              class="ant-select ant-select-outlined ant-pagination-options-size-changer css-dev-only-do-not-override-tjsggz ant-select-single ant-select-show-arrow ant-select-show-search"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-wrap"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-controls="rc_select_TEST_OR_SSR_list"
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-label="Page Size"
                      aria-owns="rc_select_TEST_OR_SSR_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="rc_select_TEST_OR_SSR"
                      role="combobox"
                      type="search"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-item"
                    title="5 /p√°g"
                  >
                    5 /p√°g
                  </span>
                </span>
              </div>
              <span
                aria-hidden="true"
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <span
                  aria-label="down"
                  class="anticon anticon-down ant-select-suffix"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </li>
        </ul>
      </div>
    </main>
  </div>
</div>
`;
